Source/Controllers/MainViewControllers/TableContent/SPTableContent.m

```diff
@@ -745,6 +745,24 @@ - (void) loadTableValues
 		[queryString appendFormat:@" ORDER BY %@", [[[dataColumns safeObjectAtIndex:[sortCol integerValue]] safeObjectForKey:@"name"] backtickQuotedString]];
 		if (isDesc) [queryString appendString:@" DESC"];
 	}
+    else // no sorting is set
+    {
+        if ([prefs boolForKey:@"SortDescendingByPrimaryKey"])
+        {
+            for (NSDictionary * column in tableDataInstance.columns)
+            {
+                if ([column[@"isprimarykey"] boolValue])
+                {
+                    NSLog(@"%@",column);
+                    [queryString appendFormat:@" ORDER BY %@ DESC",column[@"name"]];
+                    break;
+                }
+            }
+        }
+    }
```

Comment:
I'm not sure that this is the right place to hook in and make changes to sorting! Does this change here properly set the UI to match to properly display which column the data is sorted by? I believe it does not

I think it might make more sense to hook into the setTableDetails function (~line 633 "// Otherwise, clear sorting") and implement the logic to change the sortColumn there